frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,61 @@
+import { IconColor } from '#libs/enums/enums.js';
+import {
+  useCallback,
+  useHandleClickOutside,
+  useRef,
+} from '#libs/hooks/hooks.js';
+
+import { Button } from '../components.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  children: React.ReactNode;
+  title: string;
+  reference: React.RefObject<HTMLDialogElement>;
+  showCrossIcon: boolean;
+};
+
+const Modal: React.FC<Properties> = ({
+  children,
+  title,
+  reference,
+  showCrossIcon,
+}) => {
+  const modalReference = useRef<HTMLDivElement>(null);
+
+  const handleClose = useCallback(() => {
+    reference.current?.close();
+  }, [reference]);
+
+  useHandleClickOutside({
+    ref: modalReference,
+    onClose: handleClose,
+  });
+
+  return (
+    <dialog ref={reference}>
+      <div className={styles['overlay']}>
+        <div className={styles['modal']} ref={modalReference}>
+          <div className={styles['header']}>
+            <span className={styles['title']}>{title}</span>
+            {showCrossIcon && (
+              <form method="dialog" className={styles['icon-container']}>
```

Comment:
Is it form mandatory here?

Reply:
@v0ldemar01 i`ve just copied modal from [this](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/213) pr. I asked @Liza-Veis about if i should copy it and she said yes. link to [conversation](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/issues/295#issuecomment-1722203946)

Reply:
Got it, but please check with some time whether some changes are applied to original component

Reply:
Form component should be replaced with just div, it's my fault